static/src/javascripts/projects/common/modules/crosswords/crossword.js

```diff
@@ -0,0 +1,778 @@
+        const width = helpers.gridSize(this.columns);
+        const height = helpers.gridSize(this.rows);
+
+        return {
+            x: 100 * x / width,
+            y: 100 * y / height,
+        };
+    }
+
+    focusHiddenInput(x: number, y: number): void {
+        const wrapper = React.findDOMNode(
+            this.refs.hiddenInputComponent.refs.wrapper
+        );
+        const left = helpers.gridSize(x);
+        const top = helpers.gridSize(y);
+        const position = this.asPercentage(left, top);
+
+        /** This has to be done before focus to move viewport accordingly */
+        wrapper.style.left = `${position.x}%`;
+        wrapper.style.top = `${position.y}%`;
+
+        const hiddenInputNode = React.findDOMNode(
+            this.refs.hiddenInputComponent.refs.input
+        );
+
+        if (document.activeElement !== hiddenInputNode) {
+            hiddenInputNode.focus();
+        }
+    }
+
+    // Focus corresponding clue for a given cell
+    focusClue(x: number, y: number, direction: string): void {
+        const clues = helpers.cluesFor(this.clueMap, x, y);
+        const clue = clues[direction];
+
+        if (clues && clue) {
+            this.focusHiddenInput(x, y);
+
+            this.setState({
+                cellInFocus: {
+                    x,
+                    y,
+                },
+                directionOfEntry: direction,
+            });
+
+            // Side effect
+            history.replaceState(undefined, document.title, `#${clue.id}`);
+        }
+    }
+
+    // Focus first cell in given clue
+    focusFirstCellInClue(entry: Object): void {
+        this.focusClue(entry.position.x, entry.position.y, entry.direction);
+    }
+
+    focusCurrentCell(): void {
+        this.focusHiddenInput(
+            this.state.cellInFocus.x,
+            this.state.cellInFocus.y
+        );
+    }
+
+    clueInFocus(): ?Object {
+        if (this.state.cellInFocus) {
+            const cluesForCell = helpers.cluesFor(
+                this.clueMap,
+                this.state.cellInFocus.x,
+                this.state.cellInFocus.y
+            );
+            return cluesForCell[this.state.directionOfEntry];
+        }
+        return null;
+    }
+
+    allHighlightedClues(): Array<Object> {
+        return this.props.data.entries.filter(clue =>
+            this.clueIsInFocusGroup(clue)
+        );
+    }
+
+    clueIsInFocusGroup(clue: Object): boolean {
+        if (this.state.cellInFocus) {
+            const cluesForCell = helpers.cluesFor(
+                this.clueMap,
+                this.state.cellInFocus.x,
+                this.state.cellInFocus.y
+            );
+            return cluesForCell[this.state.directionOfEntry].group.includes(
+                clue.id
+            );
+        }
+        return false;
+    }
+
+    cluesData(): Array<Object> {
+        return this.props.data.entries.map(entry => {
+            const hasAnswered = helpers.checkClueHasBeenAnswered(
```

Comment:
I had to pull this out as a `const`, as there was no combination of braces and `return` that prettier would let me use to return an object literal directly from an arrow function without having some previous statement. 

I don't think it's particularly egregious to do so in this case, but the combination of errors was time consuming and frustrating to resolve.

Reply:
Have you tried running
```bash
$ make fix-commits
```
?

Reply:
I eventually managed to get it to fix it after putting it as 
```
return this.props.data.entries.map(entry => { return {
            entry,
            hasAnswered: helpers.checkClueHasBeenAnswered(
                this.state.grid,
                entry
            ),
            isSelected: this.clueIsInFocusGroup(entry),
}});
```
and prettier turning it into
```
return this.props.data.entries.map(entry => ({
            entry,
            hasAnswered: helpers.checkClueHasBeenAnswered(
                this.state.grid,
                entry
            ),
            isSelected: this.clueIsInFocusGroup(entry),
}));
```
All the other variations blew up on parsing.